asset/asset.go

```diff
@@ -2017,6 +2055,25 @@ func (req *GroupKeyRequest) Validate() error {
 			sha256.Size)
 	}
 
+	// Version 1 specific checks.
+	if req.Version == GroupKeyV1 {
+		tapscriptRoot, err := chainhash.NewHash(req.TapscriptRoot)
```

Comment:
Ah ok, it's a matter of if we'er making our tree grafting explicit or implicit...

Still gathering context, but would lean towards making it implicit (so not modifying/mutating the script tree root in place). This way it's unambiguous if a user provided a custom root or not (it'll be blank if not), and we don't need to lug around that extra field to determine if we _actually_ have a custom root or not.

Reply:
I think we have to carry around the extra custom root so that we can populate it in the group key reveal. Otherwise our peer won't be able to validate our group key.